### PR TITLE
Hcb checksinterval

### PR DESCRIPTION
**Pull request by kimobrian**
I have worked on the feature to extend check periods and grace periods to a maximum of 5 years. Please review my chnages

### CODE REVIEW

File: static/js/checks.js [hunk 1]
```diff
@@ -1,118 +1,281 @@
...
+        $('#update-name-modal').modal("show");
+        $("#update-name-input").focus();
+
+        return false;
+    });
+
+    function update_time(){
+        var entries = [0,0,0,0,0,0,0,0,0,0,0,0]
+            $period_years = $('#period_years').val()
+            if($period_years==''){
```
Comment (iamlordaubrey) [inline static/js/checks.js]: For this, the if block is not performing any action. I think it is safe to say the code can be refactored. You can have an if block that checks if $period_years/months/weeks/days/hours/mins is not an empty string, and then perform the action you're currently performing in the else block.

File: static/js/checks.js [hunk 2]
```diff
@@ -1,118 +1,281 @@
...
-        }
-    });
 
-    periodSlider.noUiSlider.on("update", function(a, b, value) {
-        var rounded = Math.round(value);
-        $("#period-slider-value").text(secsToText(rounded));
-        $("#update-timeout-timeout").val(rounded);
-    });
+            $grace_years = $('#grace_years').val()
+            if($grace_years==''){
```
Comment (iamlordaubrey) [inline static/js/checks.js]: Review comment for $period_years applies here

File: static/js/checks.js [hunk 3]
```diff
@@ -1,118 +1,281 @@
...
         }
-    });
+    })
 
-    graceSlider.noUiSlider.on("update", function(a, b, value) {
-        var rounded = Math.round(value);
-        $("#grace-slider-value").text(secsToText(rounded));
-        $("#update-timeout-grace").val(rounded);
-    });
+    $('#period_months').bind('change keyup', function(){
```
Comment (iamlordaubrey) [inline static/js/checks.js:122]: Generally, when I see repeated code, I hear a voice whispering refactor. What comes to mind is all these stuff abstracted into a function that get's called and is passed a/some parameter(s).

File: static/js/checks.js [hunk 4]
```diff
@@ -1,118 +1,281 @@
...
+        }else{
+            $('#period_minutes').css('border-color','')
+            update_time()
+        }
+    })
+
+
+        //Grace Period Errors
+
+    $('#grace_years').bind('change keyup', function(){
```
Comment (iamlordaubrey) [inline static/js/checks.js:174]: I hear the same refactor whisper

File: static/js/checks.js [hunk 5]
```diff
@@ -1,118 +1,281 @@
...
-        return false;
-    });
 
     $(".timeout-grace").click(function() {
         var $this = $(this);
 
         $("#update-timeout-form").attr("action", $this.data("url"));
-        periodSlider.noUiSlider.set($this.data("timeout"))
-        graceSlider.noUiSlider.set($this.data("grace"))
+        // periodSlider.noUiSlider.set($this.data("timeout"))
```
Comment (iamlordaubrey) [inline static/js/checks.js]: Delete if not needed

File: static/js/checks.js [hunk 6]
```diff
@@ -1,118 +1,281 @@
...
+        // graceSlider.noUiSlider.set($this.data("grace"))
         $('#update-timeout-modal').modal({"show":true, "backdrop":"static"});
 
+
+        $("#update-timeout-form").submit(event, function() {
```
Comment (iamlordaubrey) [inline static/js/checks.js]: Not sure that this guy does. Delete if not needed

**Review by iamlordaubrey — CHANGES_REQUESTED** [review]

**Comment by kimobrian** [comment]
I have made the requested changes, please review

**Review by iamlordaubrey — APPROVED** [review]

**Pull request merged**